xenon/__init__.py

```diff
@@ -63,6 +196,30 @@ def parse_args():
                               os.environ.get('BARIUM_REPO_TOKEN', ''))
     args.service_name = yml.get('service_name', 'travis-ci')
     args.service_job_id = os.environ.get('TRAVIS_JOB_ID', '')
+
+    pyproject_args = parse_pyproject(pyproject_file)
+
+    # Parse priorities - first cmd, then pyproject.toml
+    # Include args from pyproject.toml file
+    for arg_name, arg_value in pyproject_args.items():
+        # Atgument not set in pyproject.toml
+        if not arg_value:
+            continue
+
+        # Argument set from cmd
+        if getattr(args, arg_name, None) and (getattr(args_passed, arg_name, None) is not None):
+            continue
+
+        # Set only in pyproject.toml
+        setattr(args, arg_name, arg_value)
```

Comment:
Instead of cloning the args object and checking if a name is present, it's better to simply:
1. set all the values from pyproject
2. set all the values from the command line

That way no cloning is needed, no checking is needed and the priority of the sources is taken care of automatically, since CLI options will override those from pyproject.

Reply:
I am not sure if I understand it correctly. In second step when I need to get values set from cmd I also get arguments which were not passed by user but they have set default value.

Maybe if you can share piece of code to prove me where is my understanding incorrect. Then I will make necessary changes. Thank you

Reply:
@DolajoCZ We can:
1. start with the default values
2. override them with pyproject values (if any)
3. override them with command line values (if any)

Then the priority is respected naturally.

Reply:
I understand your proposed approach, but i dont understand how you want to get only arguments and values which were passed by user in cmd. If I use `args = parser.parse_args()` I get arguments and values including arguments which were not set by user. Maybe I don't know appropriate command to get from `parser` only realy passed values. Can you give me advice please?

Reply:
@DolajoCZ You can remove the defaults from the ArgumentParser object and put them in a dictionary. Then if a user does not supply an option it will be None in the result of `parser.parse_args()`. So you just skip the ones with value `None`.

Reply:
@rubik thank you for advice. This was something which does not come to my mind. I made necessary modification.

Reply:
@rubik is current code ok according your expectations or should I modify it? One thing whitch I found when I looked at other python packages witch support `pyproject.toml` is that in case when parameter can be list of values (exclude for example), then even if you want use single value you shoud use `["value"]` instead of  `"value"`. Current [implementation](https://github.com/DolajoCZ/xenon/blob/652208b8f2a3d167518884b6d93ddb2f0a694ee2/xenon/__init__.py#L57)  allow use `"value"`.

Reply:
@rubik any comment?